File: web/docs-examples/typescript/examples/embeddings.ts
/**
 * Generate text embeddings using an embedding model.
 *
 * @example
 * ```bash
 * OPENAI_API_KEY=sk-... npx tsx examples/embeddings.ts
 * ```
 */
import "@prompty/openai";
import { invoke } from "@prompty/core";
import { resolve } from "node:path";

const promptyFile = resolve(import.meta.dirname, "../../prompts/embedding.prompty");

export async function generateEmbedding(text?: string): Promise<number[]> {
  const result = await invoke(promptyFile, {
    text: text ?? "Prompty is a prompt asset format",
  });
  return result as number[];
}

// Run directly
const embedding = await generateEmbedding();
console.log(`Embedding dimensions: ${embedding.length}`);
console.log(`First 5 values: [${embedding.slice(0, 5).join(", ")}]`);

File: web/docs-examples/python/examples/embeddings.py
"""Generate embeddings with OpenAI.

This example loads an embedding .prompty file and generates a vector.
Used in: how-to/embeddings.mdx
"""
from __future__ import annotations

from prompty import invoke, load

agent = load("embedding.prompty")
vector = invoke(agent, inputs={"text": "Prompty is a prompt asset format"})
print(f"Embedding dimensions: {len(vector)}")
print(f"First 5 values: {vector[:5]}")
